galley/pkg/server/server.go

```diff
@@ -58,6 +58,7 @@ type Server struct {
 type patchTable struct {
 	logConfigure                func(*log.Options) error
 	newKubeFromConfigFile       func(string) (kube.Interfaces, error)
+	createResourceTypes         func(kube.Interfaces) error
```

Comment:
We'll have two sources of truth for CRD creation and verification: (1) the externally defined YAML files and (2) galley's k8s metadata (see https://github.com/istio/istio/blob/master/galley/pkg/metadata/kube/types.go). It might be good to plan on converging to a single source. 

Longer term I think the k8s metadata might be derived (i.e. codegen'd) from istio/api itself and we'd end up with some metadata yaml (e.g. https://github.com/istio/istio/blob/master/galley/tools/gen-meta/metadata.yaml) that can feed into CRD creation, verification, and the other parts of Galley.

Reply:
Agreed

Reply:
agreed.  while this isn't perfect, for 1.1, this should be sufficient to meet the endpoint goals of this work.  Need to see if we can get Helm to forget about CRD resources.  Then the manifests will have two components.  See [relevant issue tracker](https://github.com/istio/istio/issues/9884) for more details.